pkg/runtime/runtime.go

```diff
@@ -2318,16 +2318,20 @@ func (a *DaprRuntime) publishMessageGRPC(ctx context.Context, msg *pubsubSubscri
 	return fmt.Errorf("unknown status returned from app while processing pub/sub event %v: %v", cloudEvent[pubsub.IDField], res.GetStatus())
 }
 
-func extractCloudEventExtensions(cloudEvent map[string]interface{}) (*structpb.Struct, error) {
+func extractCloudEventExtensions(cloudEvent map[string]interface{}, metadata map[string]string) (*structpb.Struct, error) {
 	// Assemble Cloud Event Extensions:
 	// Create copy of the cloud event with duplicated data removed
+	// Merge in the pubsub publish metadata with prefix _metadata_
 
 	extensions := map[string]interface{}{}
 	for key, value := range cloudEvent {
 		if !cloudEventDuplicateKeys.Has(key) {
 			extensions[key] = value
 		}
 	}
+	for key, value := range metadata {
+		extensions[fmt.Sprintf("_metadata_" + key)] = value
```

Comment:
```suggestion
		extensions["_metadata_" + key] = value
```

Reply:
I had already changed that before you left this comment :)